aioesphomeapi/reconnect_logic.py

```diff
@@ -48,7 +51,7 @@ def __init__(
         self._cli = client
         self.name = name
         self._on_connect_cb = on_connect
-        self._on_disconnect_cb = on_disconnect
+        self._on_disconnect_cb = self._make_disconnect_cb(on_disconnect)
```

Comment:
Why do we need to be backwards compatible? We can just note a breaking change in the release notes and update it in HA when we update the lib ?

Reply:
I guess I can adjust the cli as well at the same time

Reply:
I'll take out the back compat and do a pr to esphome or as well